pilot/pkg/proxy/envoy/watcher_test.go

```diff
@@ -107,13 +107,21 @@ func TestWatchCerts_Multiple(t *testing.T) {
 }
 
 func TestWatchCerts(t *testing.T) {
-	name, err := ioutil.TempDir(os.TempDir(), "certs")
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "certs")
 	if err != nil {
-		t.Errorf("failed to create a temp dir: %v", err)
+		t.Fatalf("failed to create a temp dir: %v", err)
+	}
+	// create a temp file
+	tmpFile, err := ioutil.TempFile(tmpDir, "test.file")
+	if err != nil {
+		t.Fatalf("failed to create a temp file in testdata/certs: %v", err)
 	}
 	defer func() {
-		if err := os.RemoveAll(name); err != nil {
-			t.Errorf("failed to remove temp dir: %v", err)
+		if err := tmpFile.Close(); err != nil {
+			t.Fatalf("failed to close file %s: %v", tmpFile.Name(), err)
```

Comment:
Since these are in the defer, they can just be `Errorf` instead. TBH, you can even just ignore the errors with `_ = ...`